CryptoBook/utils.py

```diff
@@ -124,6 +124,8 @@
         responses = await asyncio.gather(
             *[ex.fetch_ohlcv(symbol, timeframe, time) for time in times]
         )
+        # Must close connection with market if using ccxt asynchronously.
+        ex.close()
```

Comment:
I believe this has to be called with the `await` symbol from my previous experiences. Any issues simply importing and running the function?

Reply:
Another note: Required would be `if not cfbypass`, since it is only without bypassing CloudFlare that we use the asynchronous functionality of the function. Otherwise `ccxt` makes no mention of closing market connection when running the function synchronously.

Reply:
regarding your first point, the error message doesn't mention that `ex.close()` must be called with the `await` symbol. 

Regarding your second point, `ex.close()` is still evaluated only if `cfbypass == False`: 

```python
if cfbypass:
        responses = [ex.fetch_ohlcv(symbol, timeframe, time) for time in times]
    else:
        responses = await asyncio.gather(
            *[ex.fetch_ohlcv(symbol, timeframe, time) for time in times]
        )
        # Must close connection with market if using ccxt asynchronously.
        ex.close()
```

Reply:
I used this as reference: https://github.com/ccxt/ccxt/blob/680bf56caa24e1d0c1ff34cf835a2e124b4d580b/examples/py/bypass-cloudflare-with-aiocfscrape.py